### PR TITLE
Add Gleam lexer

### PR DESCRIPTION
**Pull request by darenc**
Hi, this PR adds a lexer for [Gleam](https://gleam.run/), which just released version `1.0.0`. 

I've run tests and things look good to me. However, I'm not a Python person and haven't tried writing a lexer before, particularly not one for Pygments, so I'm happy to take feedback and suggestions for improvements.

There is some example HTML at https://red-rose.com/gleam.html that I generated with `pygmentize` and then modified to add the `lightbulb` style (slightly modified for Name colours).  This is the same large example file that's included with the PR.

Cheers,
Daren

### CODE REVIEW

**Comment by jeanas** [comment]
(Pushed a commit to fix line endings, can you configure Git's [core.autocrlf](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration) for the future? Thanks.)

**Comment by darenc** [comment]
> (Pushed a commit to fix line endings, can you configure Git's [core.autocrlf](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration) for the future? Thanks.)

Thank you, I thought I'd fixed them.  Sorry for the inconvenience

File: pygments/lexers/gleam.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            # Comments
+            (r'(///.*?)(\n)', bygroups(String.Doc, Whitespace)),
+            (r'(//.*?)(\n)', bygroups(Comment.Single, Whitespace)),
+
+            # Keywords
+            (keywords, Keyword),
+            (r'([a-zA-Z_]+)(\.)', bygroups(Keyword, Punctuation)),
+
+            # Punctuation
+            (r'[()\[\]{}:;,@]', Punctuation),
```
Comment (Anteru) [inline pygments/lexers/gleam.py:47]: Add a trailing `+` so you can match more than one punctuation character.

**Review by Anteru — CHANGES_REQUESTED** [review]

File: pygments/lexers/gleam.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    """
+    Lexer for the Gleam programming language (version 1.0.0).
+    """
+
+    name = 'Gleam'
+    url = 'https://gleam.run/'
+    filenames = ['*.gleam']
+    aliases = ['gleam']
+    mimetypes = ['text/x-gleam']
+    version_added = '2.17'
```
[the remark this answers is not in the document]
Comment (darenc) [reply pygments/lexers/gleam.py]: Updated, thanks

File: pygments/lexers/gleam.py (hunk 1 above)
Comment (darenc) [reply pygments/lexers/gleam.py]: Added, thanks

File: pygments/lexers/gleam.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            # Strings
+            (r'"(\\"|[^"])*"', String),
+
+            # Identifiers
+            (r'\b(let)(\s+)(\w+)', bygroups(Keyword, Whitespace, Name.Variable)),
+            (r'\b(fn)(\s+)(\w+)', bygroups(Keyword, Whitespace, Name.Function)),
+            (r'[a-zA-Z_/]\w*', Name),
+
+            # numbers
+            (r'(\d+((_\d+)+)?\.(?!\.)(\d+((_\d+)+)?)?|\.\d+((_\d+)+)?)([eEf][+-]?[0-9]+)?', Number.Float),
```
Comment (jeanas) [inline pygments/lexers/gleam.py:62]:
```suggestion
            (r'(\d+(_\d+)*\.(?!\.)(\d+(_\d+)*)?|\.\d+(_\d+)*)([eEf][+-]?[0-9]+)?', Number.Float),
```

`(X+)?` is just `X*`.

File: pygments/lexers/gleam.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            (r'\d+((_\d+)+)?[eEf][+-]?[0-9]+', Number.Float),
```
Comment (jeanas) [inline pygments/lexers/gleam.py:63]:
```suggestion
            (r'\d+(_\d+)*[eEf][+-]?[0-9]+', Number.Float),
```

File: pygments/lexers/gleam.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            (r'0[xX][a-fA-F0-9]+((_[a-fA-F0-9]+)+)?(\.([a-fA-F0-9]+((_[a-fA-F0-9]+)+)?)?)?p[+-]?\d+', Number.Float),
```
Comment (jeanas) [inline pygments/lexers/gleam.py:64]:
```suggestion
            (r'0[xX][a-fA-F0-9]+(_[a-fA-F0-9]+)*(\.([a-fA-F0-9]+(_[a-fA-F0-9]+)*)?)?p[+-]?\d+', Number.Float),
```

File: pygments/lexers/gleam.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            (r'0[bB][01]+((_[01]+)+)?', Number.Bin),
```
Comment (jeanas) [inline pygments/lexers/gleam.py:65]:
```suggestion
            (r'0[bB][01]+(_[01]+)*', Number.Bin),
```

File: pygments/lexers/gleam.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+            (r'0[oO][0-7]+((_[0-7]+)+)?', Number.Oct),
```
Comment (jeanas) [inline pygments/lexers/gleam.py:66]:
```suggestion
            (r'0[oO][0-7]+(_[0-7]+)*', Number.Oct),
```

File: pygments/lexers/gleam.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+            (r'0[xX][a-fA-F0-9]+((_[a-fA-F0-9]+)+)?', Number.Hex),
```
Comment (jeanas) [inline pygments/lexers/gleam.py:67]:
```suggestion
            (r'0[xX][a-fA-F0-9]+(_[a-fA-F0-9]+)*', Number.Hex),
```

File: pygments/lexers/gleam.py [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+            (r'\d+((_\d+)+)?', Number.Integer),
```
Comment (jeanas) [inline pygments/lexers/gleam.py:68]:
```suggestion
            (r'\d+(_\d+)*', Number.Integer),
```

**Review by jeanas — CHANGES_REQUESTED** [review]

**Comment by darenc** [comment]
> @darenc Hi, are you still working on this?

Yep, I'll pick up the requested changes in the next few days. Just been busy with other things.

File: pygments/lexers/gleam.py (hunk 9 above)
Comment (darenc) [reply pygments/lexers/gleam.py]: This and other suggestions have been implemented, thanks for the input

**Comment by Anteru** [comment]
Merged, thanks!

**Comment by Anteru** [comment]
That's fixed as well now, thanks!

**Pull request merged**